src/Detector/Ruby/RubyDetector.cs

```diff
@@ -0,0 +1,177 @@
+// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Oryx.Common.Extensions;
+using System;
+using System.Linq;
+
+namespace Microsoft.Oryx.Detector.Ruby
+{
+    /// <summary>
+    /// An implementation of <see cref="IPlatformDetector"/> which detects Ruby applications.
+    /// </summary>
+    public class RubyDetector : IRubyPlatformDetector
+    {
+        private readonly ILogger<RubyDetector> _logger;
+        private readonly DetectorOptions _options;
+
+        /// <summary>
+        /// Creates an instance of <see cref="RubyDetector"/>.
+        /// </summary>
+        /// <param name="logger">The <see cref="ILogger{RubyDetector}"/>.</param>
+        /// <param name="options">The <see cref="DetectorOptions"/>.</param>
+        public RubyDetector(ILogger<RubyDetector> logger, IOptions<DetectorOptions> options)
+        {
+            _logger = logger;
+            _options = options.Value;
+        }
+
+        public PlatformDetectorResult Detect(DetectorContext context)
+        {
+            bool isRubyApp = false;
+            string appDirectory = string.Empty;
+            var sourceRepo = context.SourceRepo;
+
+            if (sourceRepo.FileExists(RubyConstants.GemFileName))
```

Comment:
I'm not sure if we want to detect all folders? If so, I'll change it

Reply:
I think we should detect it at the root only since it's a file containing lots of dependencies, similar to `requirement.txt` to a python app, it's supposed to be at the root.